packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -51,59 +52,109 @@ export function makeVirtualReferenceManager(
    * it had been exported, we also inform the kernel that the vref has been
    * retired, so other vats can delete their weak collection entries too.
    *
-   * @param {string} vobjID  The vref of the virtual object that's plausibly dead
+   * @param {string} baseRef  The virtual object cohort that's plausibly dead
    *
-   * @returns {[boolean, boolean]} A pair of flags: the first is true if this
-   *    possibly created a new GC opportunity, the second is true if the object
-   *    should now be regarded as unrecognizable
+   * @returns {[boolean, string[]]} A pair of a flag that's true if this
+   *    possibly created a new GC opportunity and an array of vrefs that should
+   *    now be regarded as unrecognizable
    */
-  function possibleVirtualObjectDeath(vobjID) {
-    const refCount = getRefCount(vobjID);
-    const exportStatus = getExportStatus(vobjID);
-    if (exportStatus !== 'reachable' && refCount === 0) {
-      let doMoreGC = deleteStoredRepresentation(vobjID);
-      syscall.vatstoreDelete(`vom.rc.${vobjID}`);
-      syscall.vatstoreDelete(`vom.es.${vobjID}`);
-      doMoreGC = doMoreGC || ceaseRecognition(vobjID);
-      return [doMoreGC, exportStatus !== 'none'];
+  function possibleVirtualObjectDeath(baseRef) {
+    const refCount = getRefCount(baseRef);
+    const [reachable, retirees] = getExportStatus(baseRef);
+    if (!reachable && refCount === 0) {
+      let doMoreGC = deleteStoredRepresentation(baseRef);
+      syscall.vatstoreDelete(`vom.rc.${baseRef}`);
+      syscall.vatstoreDelete(`vom.es.${baseRef}`);
+      const more = ceaseRecognition(baseRef);
+      doMoreGC = doMoreGC || more;
+      return [doMoreGC, retirees];
     }
-    return [false, false];
+    return [false, []];
   }
 
-  function getRefCount(vobjID) {
-    const raw = syscall.vatstoreGet(`vom.rc.${vobjID}`);
-    if (raw) {
-      return Number(raw);
+  /**
+   * Get information about the export status of a virtual object.
+   *
+   * @param {string} baseRef  The baseRef of the virtual object of interest.
+   *
+   * @returns {[boolean, string[]]} A pair of a flag that's true if the
+   *     indicated virtual object is reachable and an array of vrefs (to facets
+   *     of the object) that should now be regarded as unrecognizable
+   */
+  function getExportStatus(baseRef) {
+    const es = syscall.vatstoreGet(`vom.es.${baseRef}`);
+    if (es) {
+      const reachable = es.indexOf('r') >= 0;
+      const retirees = [];
+      if (!reachable) {
+        if (es === 's') {
```

Comment:
Ah, ok, this is why you needed to prohibit single-facet cohorts, because otherwise we'd retire the baseref instead of `${baseref}:0`. I consider this another argument in favor of "all virtual objects are members of a `length >= 1` cohort", but I don't think I'm going to convince you of that this week, so I'll put it aside until someone else comes along and trips over this subtlety :).

Reply:
Ah, you figured it out!  Yes, I'm glad I didn't spend time explaining in my previous comment :-)